piker/brokers/kraken.py

```diff
@@ -271,18 +454,304 @@ async def bars(
+        acc_name = 'kraken.' + client._name
+        positions, vols = await client.get_positions()
+
+        all_positions = []
+
+        for ticker, pos in positions.items():
+            norm_sym = normalize_symbol(ticker)
+            if float(vols[ticker]) != 0:
+                msg = pack_position(acc_name, norm_sym, pos, vols[ticker])
+                all_positions.append(msg.dict())
+
+        await ctx.started((all_positions, (acc_name,)))
+
+        # Get websocket token for authenticated data stream
+        # Assert that a token was actually received 
+        resp = await client.kraken_endpoint('GetWebSocketsToken', {})
+        assert resp['error'] == []
+        token = resp['result']['token']
+
+        async with (
+            ctx.open_stream() as ems_stream,
+            trio.open_nursery() as n,
+        ):
+            ## TODO: maybe add multiple accounts
+            n.start_soon(handle_order_requests, client, ems_stream)
+
+            # Process trades msg stream of ws
+            async with open_autorecon_ws(
+                'wss://ws-auth.kraken.com/',
+                fixture=subscribe,
+                token=token,
+            ) as ws:
+                async for msg in process_trade_msgs(ws):
+                    for trade in msg:
+                        # check the type of packaged message
+                        assert type(trade) == Trade
+                        # prepare and send a status update for line update
+                        trade_msg = BrokerdStatus(
+                            reqid=trade.reqid,
+                            time_ns=time.time_ns(),
+
+                            account='kraken.spot',
+                            status='executed',
+                            filled=float(trade.size),
+                            reason='Order filled by kraken',
+                            # remaining='' ## TODO: not sure what to do here.
+                            broker_details={
+                                'name': 'kraken',
+                                'broker_time': trade.broker_time
+                            }
+                        )
+                        
+                        await ems_stream.send(trade_msg.dict())
+
+                        # send a fill msg for gui update
+                        fill_msg = BrokerdFill(
+                            reqid=trade.reqid,
+                            time_ns=time.time_ns(),
+
+                            action=trade.action,
+                            size=float(trade.size),
+                            price=float(trade.price),
+                            ## TODO: maybe capture more msg data i.e fees?
+                            broker_details={'name': 'kraken'},
+                            broker_time=float(trade.broker_time)
+                        )
+                        
+                        await ems_stream.send(fill_msg.dict())
+
+
+async def stream_messages(
+    ws: NoBsWs,
+):
+    '''
```

Comment:
This is the doc str style i'm kinda fond of. It's a little less noisy imo but willing to be challenged on it of course.